### ISSUE TITLE
Exit() might not close game's process

### ISSUE THREAD

**Issue by kxlow** [MonoGame/MonoGame#5416]
Calling Exit() too soon seems to prevent the game from completely closing, as the process still lives on.

Tested on MonoGame 3.5 and 3.6.

```
using Microsoft.Xna.Framework;

// This example simply adds a red rectangle to the screen
// then updates it's position along the X axis each frame.
namespace Game1
{
	public class Game1 : Game
	{
		GraphicsDeviceManager graphics;

		public Game1()
		{
			graphics = new GraphicsDeviceManager(this);
			Content.RootDirectory = "Content";
		}

		protected override void Update(GameTime gameTime)
		{
			Exit();
		}
	}
}
```

If you run this w/o debugging ( does not happen w/ debugging oddly ), the window quickly appears and closes, but the process will still exist in the Task Manager indefinitely until you end the task.

**Comment by kxlow** [comment]
It seems to get stuck in this loop: https://github.com/MonoGame/MonoGame/blob/develop/MonoGame.Framework/Windows/WinFormsGameWindow.cs#L376

**Comment by Jjagg** [comment]
Duplicate of #3995 and fixed in #5139

**Comment by kxlow** [comment]
Yeah, that seems to fix it. Thanks!
